apis/scheduling/v1alpha1/types.go

```diff
@@ -19,9 +19,10 @@ package v1alpha1
 import (
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/scheduler-plugins/pkg/apis/scheduling"
+	"sigs.k8s.io/scheduler-plugins/apis/scheduling"
 )
 
+// +kubebuilder:object:root=true
 // +genclient
 // +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
These k8s tags can be removed I think.

Reply:
No, the deepcopy runtime.objects are needed to be created hence we need this annotation. Found it used in the plugins implementations.

Reply:
I meant to leave all codegen work on scheduling/v1alpha objs to controller-gen, and config objs to k8s.io code-generator, and hence we have kubebuilder only tags on scheduling/v1alpha objs. Now, IIUC we're mixing two code generation tools on the scheduling/v1alpha objs. Correct me if I'm wrong.

Reply:
You're right, should remove it from out of tree objects.

Done.

Reply:
revisited it again and found that we have to use the `+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object` annotation to generate the `deepcopy`. But in that case, the `controller-gen` is generating that underhood, not the `code-gen`. 

ref: https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/kubebuilder_annotation.md